### PR TITLE
Configuration File Location

### PR DESCRIPTION
**Pull request by mikebirdgeneau**
Per #58, I was fiddling around since I didn't want a headers.txt in my home directory!
I'm really quite new to Rust, so this may be a complete mess - but feel free to use / modify if useful.

For the headers.txt file, this will use either:

- File in current directory (headers.txt), or
- File in ~/.config/headers.txt

Fun project - thanks for putting this together!

### CODE REVIEW

File: src/utils.rs [hunk 1]
```diff
@@ -6,6 +8,27 @@ pub fn get_project_dirs() -> Option<ProjectDirs> {
     ProjectDirs::from("com", "ccgauche", "ytermusic")
 }
 
+/// Locate the headers.txt file:
+pub fn locate_headers_file() -> Option<PathBuf> {
+
+    // Locate the headers.txt file:
+    let header_paths: [Option<PathBuf>; 2] = [
```
Comment (ccgauche) [inline src/utils.rs:15]: You can skip the `Option<PathBuf>` and just use `PathBuf` using an unwrap or ? on the from_str

**Review by ccgauche — CHANGES_REQUESTED** [review]
Thanks a lot for the contribution. I always appreciate when people improve the app.

**Comment by thomascrha** [comment]
Could we utilise a new directory inside the `.config` i.e. `$HOME/.config/ytermusic` - this is a much more common approach - it also provides the application a place to keep any future config - I can also use this dir as part of dotfiles - I'm trying to automate the Cookie fetch with some selenium and python. This way I can keep this script in my dotfiles and bootstrap the `headers.txt`

File: src/utils.rs (hunk 1 above)
Comment (mikebirdgeneau) [reply src/utils.rs]: @ccgauche I've taken a stab at changing that - assuming that I understood what you meant.

Comment (Rustmilian) [reply src/utils.rs]: @ccgauche

Comment (Rustmilian) [reply src/utils.rs]: @ccgauche please re-review this pr.

**Review by ccgauche — CHANGES_REQUESTED** [review]
Sorry for the time I took to check your PR, I don't have much time now...

File: src/utils.rs [hunk 2]
```diff
@@ -6,6 +8,25 @@ pub fn get_project_dirs() -> Option<ProjectDirs> {
...
+pub fn locate_headers_file() -> Option<PathBuf> {
+
+    // Locate the headers.txt file:
+    let header_paths: [PathBuf; 2] = [
+        PathBuf::from_str("./headers.txt").unwrap(),
+        get_project_dirs()?.config_dir().join("headers.txt"),
+    ];
+
+    // Return the first path that exists, if any:
+    for path in header_paths.iter() {
```
[the remark this answers is not in the document]
Comment (mikebirdgeneau) [reply src/utils.rs]: Thanks for the suggestion! 
Updated in commit b81247d.

**Review by ccgauche — APPROVED** [review]

**Pull request merged**